### PR TITLE
Sumo 169842 terraform delay field

### PR DESCRIPTION
**Pull request by ndhar-sumo**
Added new field 'evaluation_delay' to terraform schema

### CODE REVIEW

File: sumologic/resource_sumologic_monitors_library_monitor.go [hunk 1]
```diff
@@ -267,6 +267,11 @@ func resourceSumologicMonitorsLibraryMonitor() *schema.Resource {
 				ValidateFunc: validation.StringInSlice([]string{"Logs", "Metrics"}, false),
 			},
 
+			"evaluation_delay": {
```
Comment (rsanbhad) [inline sumologic/resource_sumologic_monitors_library_monitor.go:270]: if this will be returned in the response from the server by default, you might also want to put `Computed: true` here so the terraform state doesn't show a diff after apply - but if you get a chance you should try it out first and see if it's needed

File: sumologic/resource_sumologic_monitors_library_monitor.go [hunk 2]
```diff
@@ -504,6 +509,7 @@ func resourceSumologicMonitorsLibraryMonitorRead(d *schema.ResourceData, meta in
 	d.Set("created_by", monitor.CreatedBy)
 	d.Set("created_at", monitor.CreatedAt)
 	d.Set("monitor_type", monitor.MonitorType)
+	d.Set("evaluation_delay", monitor.EvaluationDelay)
```
Comment (rsanbhad) [inline sumologic/resource_sumologic_monitors_library_monitor.go:512]: you might want to do something similar to what we do for timerange by calling `PositiveTimeRange` function

**Review by rsanbhad — APPROVED** [review]
some comments, overall lgtm - if you get a chance, try to test locally against nite (separate from the acceptance tests/UTs) as I find it helps catch more bugs and edge cases

File: sumologic/resource_sumologic_monitors_library_monitor.go [hunk 3]
```diff
@@ -267,6 +267,11 @@ func resourceSumologicMonitorsLibraryMonitor() *schema.Resource {
 				ValidateFunc: validation.StringInSlice([]string{"Logs", "Metrics"}, false),
 			},
 
+			"evaluation_delay": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (kumar-avijit) [inline sumologic/resource_sumologic_monitors_library_monitor.go:272]: You might want to add a `ValidateFunc` here for input validation. See `time_range` for an example.

Comment (kumar-avijit) [reply sumologic/resource_sumologic_monitors_library_monitor.go:272]: I realize you might not be able to just use `time_range`'s validation function here as you probably want to admit a broader set of values. 

If that's the case, see if you could express it as a regex. You could then hand-code a validation function as done [here](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#validatefunc).

**Review by kumar-avijit — APPROVED** [review]
one comment, but lgtm.

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 1 above)
Comment (ndhar-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go:270]: I ran a test where i created a monitor on longData (prod dep). It created successfully. What diff was expected?
See below output:
`sumologic_monitor.test_log_static_resultcount: Creating...
sumologic_monitor.test_log_static_resultcount: Creation complete after 2s [id=000000000000ADAB]`

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 2 above)
Comment (ndhar-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go:512]: For this field, "0s" is allowed - that is not positive correct?

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 3 above)
Comment (ndhar-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go:272]: There is lot of validation code (Monitor manager validators) for the delay values permitted. When we try to create a monitor using terraform, won't that validation code return error messages to user? cc @rsanbhad

Comment (ndhar-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go:272]: or maybe no, because those are REST API messages?

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 1 above)
Comment (ndhar-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go:270]: Did you mean that even if i don't specify the field in request, does the response always have it?

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 2 above)
Comment (rsanbhad) [reply sumologic/resource_sumologic_monitors_library_monitor.go:512]: In this context, positive just means without the negative sign (if you look at the implementation of PositiveTimeRange function you can see it just strips the leading "-" minus sign, if it exists)

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 1 above)
Comment (rsanbhad) [reply sumologic/resource_sumologic_monitors_library_monitor.go:270]: if you manually create the monitor with this version of the provider and try to do a `terraform apply` after a successful terraform apply, does terraform say there are changes to apply, or does it say no changes detected? if no changes detected then you're fine

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 3 above)
Comment (rsanbhad) [reply sumologic/resource_sumologic_monitors_library_monitor.go:272]: those error messages will happen on `terraform apply`, but I think Avijit is asking if we can do similar validations here - if you validate in the terraform provider, then those validations run on `terraform plan` which is a better user experience (it was raised in the past that monitors terraform could do more validation locally rather than relying on server validation errors, as those make it more difficult for users to know if their terraform configuration is correct when writing it)

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 1 above)
Comment (ndhar-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go:271]: I added "Computed: true" and also there are no changes to apply once the monitor has been created with the 1st `terraform apply`

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 3 above)
Comment (ndhar-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go]: i have added ValidateFunc

Comment (kumar-avijit) [reply sumologic/resource_sumologic_monitors_library_monitor.go]: nice!
